### ISSUE TITLE
[java]SingularField can not detect static variables

### ISSUE THREAD

**Issue by wuchiuwong** [pmd/pmd#2529]
**Affects PMD Version:** 
6.22.0+
7.0.0

**Rule:**
SingularField

**Description:**
SingularField sometimes can not detect static variables, for example: <code>private static int id;</code> in the following code sample

**Code Sample demonstrating the issue:**

```
package controller;

import java.io.Serializable;
import java.sql.Date;
import java.sql.SQLException;
import javax.faces.bean.ManagedBean;
import javax.faces.bean.SessionScoped;

@ManagedBean
@SessionScoped
public class Posts implements Serializable{
    private static JDBCConnPosts posts;
    private static String title = null;
    private static String post = null;
    private static Date date = null;
    private static String name = null;
    private static int id;
    private static JsoupHtmlParser parser = new JsoupHtmlParser();
    
    public String setAllPosts(int number) throws SQLException {
        posts = new JDBCConnPosts();
        
        id = posts.getId();
        
        switch(number) {
            case 1:
                title = posts.getTitle();
                return title;
            case 2:
                post = posts.getPost();
                return post;
            case 3:
                date = posts.getDate();
                return "date";
            case 4:
                name = posts.getName();
                return name;
            case 5:
                return "posts.getId()";
                
        }
        
        return "Couldn't find DB!";
    }
    
    public String getTitle() throws SQLException {
        title = setAllPosts(1);
        return title;
    }
    
    public String getPost() throws SQLException {
        post = setAllPosts(2);
        return post;
    }
    
    public Date getDate() throws SQLException {
        setAllPosts(3);
        return date;
    }
    
    public String getName() throws SQLException {
        name = setAllPosts(4);
        return name;
    }
}
```

**Expected outcome:**
false-negative

**Running PMD through:**
CLI

**Comment by jsotuyod** [comment]
This is intended behavior, statics and final variables are explicitly skipped, although I'm unsure the reasoning behind this. The existing tests for this behavior do not help to clarify this.

https://github.com/pmd/pmd/blob/86328b7ad16b8a73af0cf86cbedadd7595eeedea/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/SingularFieldRule.java#L98-L101

**Comment by oowekyala** [comment]
This will be fixed by #4535. The restriction to non-static fields was there to match existing PMD 6 behavior, but it is not necessary.

**Comment by oowekyala** [comment]
This is actually a duplicate of #3694. This issue is older but the other one had more activity so I'm closing this one.
